### ISSUE TITLE
Inherit Dataset into specific objects

### ISSUE THREAD

**Issue by sgibardDevelopment** [sgibardDevelopment/ML_preprocessing_tools#8]
Allow inheritance of Dataset object into : 
- a training set object
- a validation set object
- let target data be dependant of those new objects

**Comment by sgibardDevelopment** [comment]
Allow inheritance of Dataset object into : 
- Add a test set object
